packages/e2e/package.json

```diff
@@ -22,7 +22,7 @@
     "@types/chai": "^4.2.0",
     "@types/fs-extra": "^5.1.0",
     "@types/mocha": "^9.0.0",
-    "@types/node": "^12.0.0",
+    "@types/node": "^14.14.31",
```

Comment:
I think we should leave this as 12.x.y if possible, as that would be the oldest version that we support.

Reply:
Did you notice my commit message in https://github.com/nomiclabs/hardhat/pull/2346/commits/309cb4d9b62fb286f5f903a33657734762974b75 ? Without updating to 14.x (and I just reconfirmed it by reverting that commit), I get similar errors to what was mentioned in that `undici` issue; specifically:

```
yarn run v1.22.11
$ tsc --build .
../../node_modules/undici/types/dispatcher.d.ts:5:10 - error TS2305: Module '"buffer"' has no exported member 'Blob'.

5 import { Blob } from 'buffer'
           ~~~~

../../node_modules/undici/types/file.d.ts:4:10 - error TS2305: Module '"buffer"' has no exported member 'Blob'.

4 import { Blob, BlobOptions } from 'buffer'
           ~~~~

../../node_modules/undici/types/file.d.ts:4:16 - error TS2305: Module '"buffer"' has no exported member 'BlobOptions'.

4 import { Blob, BlobOptions } from 'buffer'
                 ~~~~~~~~~~~

../../node_modules/undici/types/fetch.d.ts:5:10 - error TS2305: Module '"buffer"' has no exported member 'Blob'.

5 import { Blob } from 'buffer'
           ~~~~

src/internal/hardhat-network/provider/fork/rpcToTxData.ts:27:5 - error TS2322: Type '(Buffer | Buffer[])[][] | undefined' is not assignable to type 'AccessList | AccessListBuffer | undefined'.
  Type '(Buffer | Buffer[])[][]' is not assignable to type 'AccessList | AccessListBuffer | undefined'.
    Type '(Buffer | Buffer[])[][]' is not assignable to type 'AccessList'.
      Type '(Buffer | Buffer[])[]' is missing the following properties from type 'AccessListItem': address, storageKeys

27     accessList: rpcTransaction.accessList?.map((item) => [
       ~~~~~~~~~~


Found 5 errors.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Reply:
Oh, I missed that comment!

I think I'd rather enable 

```
"skipDefaultLibCheck": true,
"skipLibCheck": true
```

in the `compilerOptions` section of https://github.com/nomiclabs/hardhat/blob/master/config/typescript/tsconfig.json — This is actually de default if you do `npx tsc --init` now.

My reasoning: an invalid dependency .d.ts is less likely to bite us than `@types/node` being out of sync with the node.js versions that we support.

As a plus, this will improve the compilation speeds of all the packages.

Reply:
Changed `tsconfig` in https://github.com/nomiclabs/hardhat/pull/2346/commits/13d12c426e35bb6d720342d1de1727ae7d718071 and reverted the `@types/node` dependency change in https://github.com/nomiclabs/hardhat/pull/2346/commits/648999c5eece1007af3a7ecceb066f9a83ec00ec